### ISSUE TITLE
When using the MSBuild package with Blazor library extra content files are added to nupkg

### ISSUE THREAD

**Issue by attilah** [microsoft/TypeScript#42907]
# Bug Report

When using the `Microsoft.Typescript.MSBuild` 4.2.1-rc to pack a Blazor Library project for nuget package, the compiled javascript files are added to the `content` and `contentFiles` folders of the nuget package, not just to `staticWebAssets` that Blazor build picks up.

`GetTypeScriptOutputForPublishing` MSBuild target seems to be the source of errors, if the content of that target is commented out, the nuget package does not have the above problem.

### Repro

1. Clone https://github.com/BlazorExtensions/Logging
2. dotnet pack
3. Observe the contents of the nupkg file

### 🔎 Search Terms

msbuild, nuget, blazor

### 🙁 Actual behavior

See in summary above.

### 🙂 Expected behavior

Nuget package should not have those extra files.

**Comment by minestarks** [comment]
Interesting issue, thanks for pointing it out. It's currently intentional that the TS NuGet package includes the output as `Content` items, but I'm not familiar enough with the design of Blazor Library projects to be able to tell whether the fault lies with the TS NuGet package or the Blazor project type here. Will investigate.

**Comment by attilah** [comment]
If we could have a condition for the `GetTypeScriptOutputForPublishing` target that we could set to true to disable the content addition that would be awesome. The problem was that this target is hooked up into multiple places and not all of them has an "After" target where I could revert these things, I tried to apply all my msbuild fu, but failed.

**Comment by galvesribeiro** [comment]
For the sake of giving more context on this, if you build the project using `dotnet build`, VSCode or Rider, the problem on the consuming projects doesn't happen since it looks like it doesn't care about the _link_ to the transpiled JS file on wwwroot. However, if you build on VS and you have the tsconfig on it as Attila mentioned, it will completely fail the build.

This is how VS 2019 sees the project:

![image](https://user-images.githubusercontent.com/4714040/108757714-f5eca380-7528-11eb-8411-30b2b34d2fc3.png)

Note that there are links on `wwwroot/BrowserConsoleLogger.js` and `tsconfig.json` which shouldn't be there since they are both internal to the NuGet package. In fact, we're tried adding `None` with `Pack=false` and the same behavior still happen.

It indeed looks like the TS package adds everything from the project as `Content` on nuget so the consumer will have it "available". If that was the intentional behavior, I think for the sake of custom builds, like Blazor (which share a lot from Razor Components project type itself), we should have a prop which flag whether or not we want that to be added as Content.

**Comment by minestarks** [comment]
@attilah so I don't know how much you're going to like this, but one workaround I found is to override the offending target by adding this to your `Directory.Build.targets`:
```
<Target Name="GetTypeScriptOutputForPublishing" />
```

Still should investigate the inner workings of this and find out how we can integrate better with Blazor though.

**Comment by attilah** [comment]
@minestarks Yeah....as a last resort that could work I intentionally did not bring it up as a possible solution :D :D :D I'd appreciate if we can fix it in a proper way. Of course as a workaround target override is fine!

**Comment by minestarks** [comment]
@attilah ah yes, you already got there before me! I can't see any downsides to your suggestion of adding a condition to optionally disable the target. This is what I'll do if there's no more elegant integration we can come up with.

**Comment by marcOcram** [comment]
Any news on this? It is still an issue with Blazor 6, .NET 6 and/or VS 2022. I just stumbled across this and took me a while to figure this out ...
